modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java

```diff
@@ -1636,6 +1644,190 @@ public void testExe() throws Exception {
+
+        res = jsonTaskResult(ret);
+
+        assertEquals("5", res.get("result").toString());
+
+        // Check composite object key.
+        CompositeKeyExternal key = new CompositeKeyExternal(new CompositeKeyInternal());
+
+        jcache().put(key, 6);
+
+        ret = content(new VisorGatewayArgument(VisorCacheGetValueTask.class)
+            .setNode(locNode)
+            .setTaskArgument(VisorCacheGetValueTaskArg.class)
+            .addArguments(DEFAULT_CACHE_NAME,
+                "{" +
+                    "\"type\":\"" + VisorDataType.BINARY + "\"," +
+                    "\"value\":{" +
+                        "\"className\":\"" + CompositeKeyExternal.class.getName() + "\"," +
+                        "\"fields\":[" +
+                            "{\"type\":\"INT\",\"name\":\"id\",\"value\":" + key.getId() + "}," +
+                            "{\"type\":\"BINARY\",\"name\":\"internal\",\"value\":{" +
+                                "\"className\":\"" + CompositeKeyInternal.class.getName() + "\"," +
+                                "\"fields\":[{\"type\":\"INT\",\"name\":\"id\",\"value\":" + key.getInternal().getId() + "}]" +
+                            "}}" +
+                        "]" +
+                    "}" +
+                "}"));
+
+        info("VisorCacheGetValueTask result for binary key: " + ret);
+
+        res = jsonTaskResult(ret);
+
+        assertEquals("6", res.get("result").toString());
+    }
+
+    /**
+     * Tests work of VisorCacheGetValueTask with unsupported key type.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testVisorGetCacheValueForUnsupportedKey() throws Exception {
+        ClusterNode locNode = grid(1).localNode();
+
+        String ret = content(new VisorGatewayArgument(VisorCacheGetValueTask.class)
+            .setNode(locNode)
+            .setTaskArgument(VisorCacheGetValueTaskArg.class)
+            .addArguments(DEFAULT_CACHE_NAME, "{\"type\":\"UNSUPPORTED\",\"value\":\"1\"}"));
+
+        info("VisorCacheGetValueTask result for unsupported key type: " + ret);
+
+        JsonNode err = jsonTaskErrorResult(ret);
+
+        assertFalse(err.isNull());
+        assertTrue(err.textValue().contains(UNSUPPORTED_KEY_TYPE));
+    }
+
+    /**
+     * Tests work of VisorCacheGetValueTask result with missed value.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testVisorGetCacheValueWithMissedValue() throws Exception {
+        ClusterNode locNode = grid(1).localNode();
+
+        String ret = content(new VisorGatewayArgument(VisorCacheGetValueTask.class)
+            .setNode(locNode)
+            .setTaskArgument(VisorCacheGetValueTaskArg.class)
+            .addArguments(DEFAULT_CACHE_NAME, "{\"type\":\"" + VisorDataType.INT + "\",\"value\":\"1\"}"));
+
+        info("VisorCacheGetValueTask result for integer key type with missed value: " + ret);
+
+        JsonNode res = jsonTaskResult(ret);
+
```

Comment:
I guess the following line is missing:
`assertFalse(err.isNull());`

Reply:
jsonTaskResult function contains check assertTrue(res.get("error").isNull()).